### DISCUSSION TITLE
Updating device's attributes after making changes to device type

### DISCUSSION THREAD

**Discussion by BriacD** [netbox-community/netbox#14525] [Q&A]
As I understand it, when a device is created and references a device type, it will create the device using the device types attributes.
I realized that I had a typo on a particular switch model and realized it did not update on all 300+ switches of this type that are in my environment.

My question is, other than maybe doing an update via the device ID, is there a way to force Netbox to inherit the changes to the type without jumping through too many hoops? I've noticed, some changes/dependencies are naturally inherited due to some backend relationships but not in this case.

I assume the answer is likely no but figured it couldn't hurt to ask. At the least, maybe I stand to gain a better understanding of these kinds of issues.

**Comment by jeremystretch** [comment]
> I realized that I had a typo on a particular switch model

On what model & field was the error made? The model is an attribute of the _device type_ model, so it would only need to be corrected there. If it's a field on the _device_ model, you can filter the list of device by type and apply the change using NetBox's bulk edit function.

**Reply by BriacD** [reply]
Thanks for the reply.

I defined my interfaces incorrectly on a WS 3560-CX-12PC-S. After updating the interface names, I noticed on a given switch of that model that the interfaces had not updated.

Separately, I attempted to update a given switch's model with the same model. Essentially:
name,device_type,id
test_switch,WS-C3560-8PC,2075

To _force_ and update. This also did not work as Netbox says WS 3560-CX-12PC-S does not exist. I have verified that it does.
I suspect I may be facing another issue with my models that is throwing me off.

**Reply by selvigp** [reply]
@BriacD Did you resolve this issue

**Reply by BriacD** [reply]
I think I ended up using the custom script: add device type components

https://github.com/netbox-community/customizations/tree/master/scripts
